autogpt_platform/frontend/src/hooks/useAgentGraph.ts

```diff
@@ -405,7 +405,34 @@ export default function useAgentGraph(
             setNestedProperty(errors, key, "This field is required");
           }
         });
-      }
+      } 
+      
+      Object.entries(node.data.inputSchema.properties || {}).forEach(([key, schema]) => {
+        if ('depends_on' in schema) {
+          const dependencies = Array.isArray(schema.depends_on) ? schema.depends_on : [schema.depends_on];
+          
+          // Check if dependent field is set while required fields are empty
+          const hasValue = inputData[key as keyof typeof inputData] && String(inputData[key as keyof typeof inputData]).length > 0;
```

Comment:
- Does this take defaults into account?
- This check breaks when the value is `false`. 

```suggestion
          const mustHaveValue = inputSchema.required && key in inputSchema.required;
          const hasValue = schema.default != null || inputData[key] != null;
```